pkg/api/v1/defaults.go

```diff
@@ -389,3 +389,31 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+func SetDefaults_SecurityContext(sc *v1.SecurityContext) {
+	ape := true
+	if sc == nil {
+		sc = &v1.SecurityContext{
+			AllowPrivilegeEscalation: &ape,
+		}
+		return
+	}
+
+	if sc.Privileged != nil && *sc.Privileged {
```

Comment:
there should be a big `if sc.AllowPrivilegeEscalation == nil {` around these... as is, it overwrites whatever the user specified